packages/react-composites/tests/browser/callwithchat/hermetic/fixture.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Page, test as base } from '@playwright/test';
+import path from 'path';
+import { createTestServer } from '../../common/server';
+import { loadNewPageWithPermissionsForCalls } from '../../common/fixtureHelpers';
+import { dataUiId, encodeQueryData, waitForPageFontsLoaded, waitForSelector } from '../../common/utils';
+import type { FakeChatAdapterArgs, MockCallAdapterState } from '../../../common';
+
+const SERVER_URL = 'http://localhost';
+const APP_DIR = path.join(__dirname, '../../../app/callwithchat');
+
+/**
+ * Create the test URL.
+ *
+ * @param serverUrl - URL of webpage to test, this is typically https://localhost:3000
+ * @param mockCallAdapterState - Initial state for the {@link MockCallAdapter} constructed by the test app.
+ * @param fakeChatAdapterArgs - Args for fake adapter setup to add to the query search parameters of the URL.
+ * @param qArgs - Optional args to add to the query search parameters of the URL.
+ *
+ * @returns URL string
+ */
+export const buildUrlForApp = (
+  serverUrl: string,
+  mockCallAdapterState: MockCallAdapterState,
+  fakeChatAdapterArgs: FakeChatAdapterArgs,
+  qArgs?: { [key: string]: string }
+): string => {
+  return `${serverUrl}?${encodeQueryData({
```

Comment:
Nothing significant to note here, a random observation: 
URLs have a max length of 2048 characters. So loading a URL might experience odd behaviour if there are multiple users with really long usernames in our tests. Which we can easily avoid and constrain if we wanted to.

Reply:
Yeah, we'll cross that bridge when we have to.

gzip ftw ;)